### PR TITLE
Upgrade versions

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,10 @@
 # Changelog
 
-## Unreleased
+## 0.2.0.2 (Mar 2023)
 
 * `toChunks` and `toBytes` now make the stdin available to the process being
```
Comment (harendra-kumar) [inline CHANGELOG.md:5]: Isn't this a breaking change?
Can you diff the previous release using packdiff and see what other changes are there in this release?

File: cabal.project.user [hunk 2]
```diff
@@ -3,12 +3,12 @@ packages: streamly-process.cabal
 source-repository-package
    type: git
    location: https://github.com/composewell/streamly.git
-   tag: master
+   tag: 24926eb041da4bc25c55460e0ecbf18fcb33e990
```
Comment (harendra-kumar) [inline cabal.project.user:6]: This is an old commit of streamly. We should not use source-repository stanza, instead allow cabal to use it from hackage.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Comment by rnjtranjan** [comment]
---------------------------------
Printing diff
---------------------------------

[C] Streamly.System.Process
    [C] toChunks
        [O] toChunks :: (IsStream t, MonadAsync m, MonadCatch m) => FilePath -> [String] -> t m (Array Word8)
        [N] toChunks :: (MonadAsync m, MonadCatch m) => FilePath -> [String] -> Stream m (Array Word8)
    [C] toBytes
        [O] toBytes :: (IsStream t, MonadAsync m, MonadCatch m) => FilePath -> [String] -> t m Word8
        [N] toBytes :: (MonadAsync m, MonadCatch m) => FilePath -> [String] -> Stream m Word8
    [D] processChunks :: (MonadCatch m, MonadAsync m) => FilePath -> [String] -> Stream m (Array Word8) -> Stream m (Array Word8)
    [D] processBytes :: (MonadCatch m, MonadAsync m) => FilePath -> [String] -> Stream m Word8 -> Stream m Word8
    [A] pipeChunks :: (MonadCatch m, MonadAsync m) => FilePath -> [String] -> Stream m (Array Word8) -> Stream m (Array Word8)
    [A] pipeBytes :: (MonadCatch m, MonadAsync m) => FilePath -> [String] -> Stream m Word8 -> Stream m Word8

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,10 @@
 # Changelog
 
-## Unreleased
+## 0.3.0.0 (Mar 2023)
 
 * `toChunks` and `toBytes` now make the stdin available to the process being
   executed.
+* Allow streamly 0.9.0
```
Comment (adithyaov) [inline CHANGELOG.md:7]: There are some breaking changes. And deprecations that probably need to be mentioned here.
Also, API changelog can be added.

File: streamly-process.cabal [hunk 4]
```diff
@@ -1,6 +1,6 @@
 cabal-version:       2.2
 name:                streamly-process
-version:             0.2.0.1
+version:             0.3.0.0
```
Comment (adithyaov) [inline streamly-process.cabal:3]: Do we need the patch version here? Just `0.3.0`?

File: default.nix [hunk 5]
```diff
@@ -36,10 +36,10 @@ let
                   utils.local super "streamly-process" ./. "--benchmark" inShell;
                   streamly = utils.composewell super
                       "streamly"
-                      "8240f5f870fe47623df99514aed6a542f80c9641";
+                      "9596aea3ab33d3c2fa513247fcfe8604bd7ff454";
```
Comment (adithyaov) [inline default.nix:39]: We can get streamly and streamly core from hackage instead. They are released.

File: src/Streamly/Internal/System/Process.hs [hunk 6]
```diff
@@ -137,6 +138,7 @@ import qualified Streamly.Internal.FileSystem.Handle
     as Handle (readChunks, putChunks)
 import qualified Streamly.Unicode.Stream as Unicode
 import qualified Streamly.Internal.Unicode.Stream as Unicode (lines)
+import Data.Kind (Type)
```
Comment (adithyaov) [inline src/Streamly/Internal/System/Process.hs:141]: Do we want to make this a must?
We can add a no warning in the cabal file instead. This is tedious to write for every function.

File: .github/workflows/haskell.yml [hunk 7]
```diff
@@ -110,20 +110,6 @@ jobs:
...
-            cabal-version: 3.2
-            cabal_project: cabal.project.user
-
-          - name: 8.6.5
-            ghc_version: 8.6.5
-            build: cabal-v2
-            runner: ubuntu-latest
-            cabal-version: 3.2
-            cabal_project: cabal.project.user
-
```
Comment (adithyaov) [inline .github/workflows/haskell.yml:126]: Why remove these?

File: streamly-process.cabal [hunk 8]
```diff
@@ -16,11 +16,9 @@ copyright:           Composewell Technologies
 category:            Streamly, Streaming, System
 stability:           Experimental
 tested-with:         GHC==9.4.2
-                   , GHC==9.2.1
+                   , GHC==9.2.2
```
Comment (adithyaov) [inline streamly-process.cabal:19]: This should match the tested versions in the CI

File: test/Streamly/System/Process.hs [hunk 9]
```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE StandaloneKindSignatures #-}
```
Comment (adithyaov) [inline test/Streamly/System/Process.hs:2]: Remove

File: test/Streamly/System/Process.hs [hunk 10]
```diff
@@ -35,7 +36,9 @@ import qualified Streamly.System.Process as Proc
 import qualified Streamly.Internal.FileSystem.Handle as FH (putBytes, read)
 import qualified Streamly.Internal.System.Process as Proc
     (pipeChunks', pipeBytes', toChunks', toBytes')
+import Data.Kind (Type)
```
[the remark this answers is not in the document]
Comment (rnjtranjan) [reply test/Streamly/System/Process.hs:39]: It gives another warning for lower version of ghc: **unrecognised warning flag: -Wno-missing-kind-signatures**

File: streamly-process.cabal (hunk 8 above)
Comment (rnjtranjan) [reply streamly-process.cabal:19]: Yes, CI is for 9.2.2.

File: test/Streamly/System/Process.hs (hunk 9 above)
Comment (rnjtranjan) [reply test/Streamly/System/Process.hs:2]: Appveyor CI will fail.

File: .github/workflows/haskell.yml (hunk 7 above)
Comment (rnjtranjan) [reply .github/workflows/haskell.yml:126]: option "StandaloneKindSignatures" is not supported in these versions of ghc.

File: test/Streamly/System/Process.hs (hunk 10 above)
[the remark this answers is not in the document]
Comment (harendra-kumar) [reply test/Streamly/System/Process.hs:39]: We do not care about warnings in lower versions of ghc.

File: .github/workflows/haskell.yml (hunk 7 above)
Comment (harendra-kumar) [reply .github/workflows/haskell.yml:126]: this is an important package and we should support all versions of GHC that streamly supports. We do not need or use StandaloneKindSignatures .

File: streamly-process.cabal [hunk 11]
```diff
@@ -50,6 +50,15 @@ flag use-native
...
 common compile-options
   default-language: Haskell2010
+  if os(darwin)
+    cpp-options:    -DCABAL_OS_DARWIN
+
+  if os(linux)
+    cpp-options:    -DCABAL_OS_LINUX
+
+  if os(windows)
+    cpp-options:    -DCABAL_OS_WINDOWS
```
Comment (harendra-kumar) [inline streamly-process.cabal:60]: Should not have to define our own host macros via cabal and pass them when GHC anyway supplies such macros.

For windows  `mingw32_HOST_OS`  is always defined by GHC.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Pull request merged**